src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -323,270 +304,210 @@ std::vector< T > gatherData( std::vector< T > const & data )
 }
 
 /**
- * @brief This method is used to preprocess the VTK mesh and count the number of cells, facets, regions and
- * surfaces over the current MPI rank
- * @param[in] mesh the vtkUnstructuredGrid that is loaded
- * @param[out] regionsHex map from region index to the hexahedron indexes
- * @param[out] regionsTetra map from region index to the tetra indexes
- * @param[out] regionsWedges map from region index to the wedges indexes
- * @param[out] regionsPyramids map from region index to the pyramids indexes
- * @return A map from the surface indices to the associated cell ids for the current rank.
- * The mapping contains the surface indices for the whole simulation across the MPI ranks,
- * even if there is no cell for current rank (then the list will be empty).
+ * @brief Get the GEOSX element type
+ * @param[in] cellType The vtk cell type
+ * @return The GEOSX element type
  */
-std::map< int, std::vector< vtkIdType > > buildRegionToCellsAndFaces( vtkSmartPointer< vtkUnstructuredGrid > mesh,
-                                                                      std::map< int, std::vector< vtkIdType > > & regionsHex,
-                                                                      std::map< int, std::vector< vtkIdType > > & regionsTetra,
-                                                                      std::map< int, std::vector< vtkIdType > > & regionsWedges,
-                                                                      std::map< int, std::vector< vtkIdType > > & regionsPyramids )
+ElementType convertVtkToGeosxElementType( VTKCellType const cellType )
```

Comment:
If modifying this function, performances could be drastically  improved by first counting the number of each ElementType, then allocating, and finally filling.
Is a std::map really needed ? (performances are never the best)  Why not 2 vectors ?

Reply:
Indeed, I will improve this.

Originally I wasn't sure about the overhead of double-traversal of `vtkUnstructuredGrid`, because for example `mesh.GetCellType()` is a non-final virtual method (i.e. the call cannot be de-virtualized by the compiler) and we're calling it millions of times in a hot loop - do we then want a second loop? But I think indirect branch predictor in the CPU saves us there, it's unlikely to be any worse that reallocations (even amortized).